### PR TITLE
Search and pagination

### LINKED ISSUE #19: Support Client Side Search Functionality
Create a search box, highlight and expand if necessary on the text that user input in the box.

### LINKED ISSUE #20: Support Client-side Pagination
By default, show rows sorted by the last updated timestamp, and only display the first N rows, create a button to show more data.

### PR DESCRIPTION
**Pull request by sasymer**
- Combine Millennia's search with pagination
- Make sure search works with pagination and sort

### CODE REVIEW

File: src/main/webapp/table.js [hunk 1]
```diff
@@ -139,10 +156,46 @@ class Table {
...
+    for (index in this.dataTable) {
+      const row = this.dataTable[index];
+
+      let rowIndex;
+      for (rowIndex in row) {
+        let rowData = row[rowIndex];
+        if (rowData === null || rowData === "") {
+          //No data for this row
+        } else {
+          rowData = rowData.toString(); //converting data type to string for comparison
```
Comment (wdcroslen) [inline src/main/webapp/table.js]: Can combine these two lines to be rowData = rowData.toString().toLowerCase();

File: src/main/webapp/table.js [hunk 2]
```diff
@@ -139,10 +156,46 @@ class Table {
       }
       table.appendChild(rowElement);
     }
-    const pageInfoString = "Displaying rows " + rowToStart + " to " + rowToEnd + " of " + this.dataTable.length;
+    const pageInfoString = "Displaying rows " + rowToStart + " to " + rowToEnd + " of " + this.filteredRows.length;
     return pageInfoString;
   }
 
+  //Filters data by user input on the search bar
+  getFilteredRows(){
```
Comment (wdcroslen) [inline src/main/webapp/table.js:164]: I really like this method, the variable names and comments make it easy to follow!

**Review by wdcroslen — APPROVED** [review]
Changes look great, I really want to see this in action!

**Review by jiaxinz-g — APPROVED** [review]

File: src/main/webapp/table.js (hunk 2 above)
Comment (sasymer) [reply src/main/webapp/table.js:164]: It's millennia's @hilakey !

**Pull request merged**